### PR TITLE
net: lib: ptp: Include timestamp in follow up msg and minor fixes

### PR DESCRIPTION
**Pull request by tomasixys**
This PR aims to fix some small issues encountered while implementing ptp on a stm32f767. The fixes are described in the individual commit messages, but below is a brief description:

While in developement, I have asserts enabled, and experienced the asserts in transport.c always failing. It appears the condition is inverted, as PTP_SOCKET_CNT is always larger than the current socket.
The second issue is tied to the ptp_clock_synchronization in clock.c. Casting the adjustment value to a uint64/32 with a compound assignment operator (-=) causes the adjustment to be mono-directional.
The third issue is due to compilation errors occuring due to some reserved variable names are being used.
The fourth and main issue is how the follow up message lacks the timestamp captured at the sync message egress. The timestamp is prepared by the sync message, but not inserted into the follow up.

### CODE REVIEW

**Review by kartben — APPROVED** [review]
Thanks, this looks good to me!
For second issue, I vaguely remember that I was indeed possibly not fixing this properly when taking a stab at fixing the previously broken implementation, but since the problem wasn't gonna arise in my particular use case, I probably forgot to invest more time in a proper fix 🫣

File: subsys/net/lib/ptp/msg.c
Comment (awojasinski) [inline subsys/net/lib/ptp/msg.c:1]: Timestamp in follow up is taken from a sync message. Please take a look at the `port_sync_timestamp_cb()` function in `port.c:204`. The follow up doesn't need it's own timestamp.

**Review by awojasinski — CHANGES_REQUESTED** [review]

File: subsys/net/lib/ptp/msg.c
Comment (tomasixys) [reply subsys/net/lib/ptp/msg.c:1]: This is correct, but the timestamp is not processed correctly. The port_sync_timestamp_cb() *prepares* the timestamp for the follow up message, but the time stamp is not converted to correct byteorder without being processed by msg_timestamp_pre_send().

File: subsys/net/lib/ptp/transport.c [hunk 1]
```diff
@@ -266,7 +266,7 @@ int ptp_transport_close(struct ptp_port *port)
 
 int ptp_transport_send(struct ptp_port *port, struct ptp_msg *msg, enum ptp_socket idx)
 {
-	__ASSERT(PTP_SOCKET_CNT <= idx, "Invalid socket index");
+	__ASSERT(PTP_SOCKET_CNT > idx, "Invalid socket index");
```
[the remark this answers is not in the document]
Comment (awojasinski) [reply subsys/net/lib/ptp/transport.c:269]: `PTP_SOCKET_CNT` isn't valid socket id. Each port has at the moment 2 sockets - idx 0 for events and idx 1 for general. The `PTP_SOCKET_CNT` is used when specifying number of sockets (n_ports * `PTP_SOCKET_CNT`) so yes, we want to assert if idx is <= PTP_SOCKET_CNT

File: subsys/net/lib/ptp/msg.c
Comment (awojasinski) [reply subsys/net/lib/ptp/msg.c:1]: sorry you're right I overlooked that

File: subsys/net/lib/ptp/transport.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomasixys) [reply subsys/net/lib/ptp/transport.c:269]: as we agree that PTP_SOCKET_CNT is *not* a valid socket id, then should we not keep the comparison to be *strictly* less than: 
    idx < PTP_SOCKET_CNT

[the remark this answers is not in the document]
Comment (awojasinski) [reply subsys/net/lib/ptp/transport.c:269]: yes, sorry

**Review by awojasinski — APPROVED** [review]

**Pull request merged**